src/Simplex/Messaging/Agent.hs

```diff
@@ -527,7 +532,14 @@ deleteConnectionsAsync_ onSuccess c connIds = case connIds of
 switchConnectionAsync' :: AgentMonad m => AgentClient -> ACorrId -> ConnId -> m ()
 switchConnectionAsync' c corrId connId =
   withStore c (`getConn` connId) >>= \case
-    SomeConn _ DuplexConnection {} -> enqueueCommand c corrId connId Nothing $ AClientCommand $ APC SAEConn SWCH
+    SomeConn _ (DuplexConnection _ rqs@(rq :| _rqs) _) -> case findSwitchedRQ rqs of
+      Nothing -> do
+        let queueNotSwitched = \RcvQueue {switchStatus} -> isNothing switchStatus
+            updateSwitchStatus = \db q -> setRcvQueueSwitchStatus db q (Just RSSQueueingSwch)
+        withExistingRQ c rq queueNotSwitched updateSwitchStatus >>= \case
+          Right _replaced' -> enqueueCommand c corrId connId Nothing $ AClientCommand $ APC SAEConn SWCH
+          Left qce -> switchCannotProceedErr $ qceStr qce "queue expected to be not switched"
```

Comment:
this is just store operation, no network IO - why can't we just update the status after it's been enqueued and do it in a single transaction? the status then will be RSSwchQueued

Reply:
to prohibit queueing second SWCH command or simultaneously calling synchronous function, though it may be a bit redundant